src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryDate.cs

```diff
@@ -405,7 +405,11 @@ private static FormulaValue DateParse(EvalVisitor runner, EvalVisitorContext con
         public static FormulaValue DateTimeParse(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, StringValue[] args)
         {
             var str = args[0].Value;
-            if (DateTime.TryParse(str, runner.CultureInfo, DateTimeStyles.None, out var result))
+
+            // argCI will have Culrural info incase one was passed in argument else it will have the default one.
+            CultureInfo argCI = args.Length > 1 ? new CultureInfo(args[1].Value) : runner.CultureInfo;
```

Comment:
>new CultureInfo(args[1].Value) [](http://example.com/codeflow?start=50&length=30)

We should validate this parameter. 
eg, what if the maker passes an invalid value ("xxxx")

Reply:
This should probably be ion a helper function since I suspect other places may want to call it. 

eg, do we have a similar issue for TimeParse?

Reply:
We'll also have this concern for Text, Value,...

Reply:
@MikeStall yes, the text function also has an issue,
e.g. the below doesn't work correctly!
Text(1234567.89, "[$-fr-FR]# ###,## €", "fr-FR")
Should I fix this one also in current fix?

Reply:
Sure - if it's easy to do. We'll just get them all at once. 
If it's not easy, we can do in a future change.

Reply:
@MikeStall The issue with Text seems to be a different one, it does take culture into account, but the bug is with "format" of the string. I've updated the Value, TimeValue, And DateTimeValue for now.